psycopg3/psycopg3/proto.py

```diff
@@ -51,14 +51,12 @@
 
 
 class RowMaker(Protocol):
-    def __call__(self, __values: Sequence[Any]) -> Row:
+    def __call__(self, __values: Sequence[Any]) -> Any:
```

Comment:
There was indeed a problem with this this type definition, as evidenced by mypy reporting an invalid argument type in `psycopg3.connect(row_factory=dict_row)` for instance. Not sure we can do better than `Any`, though... Or maybe this is a misusage of `TypeVar`? 

(Just thinking out load now, I keep thinking about that...)

Reply:
I think we are stretching what mypy can do here. I have noticed that if you change `Row` definition it explodes a bit... I wouldn't dare touching it.

I suspect the cursor should be defined as a Generic on the Row too, but so the Connection should, and because the row can be changed at runtime maybe it's just not a thing you can do :)